modules/programs/ssh.nix

```
@@ -282,7 +291,15 @@ let
   });
 
   matchBlockStr = cf: concatStringsSep "\n" (
-    ["Host ${cf.host}"]
+    (
+      if cf.exec != null
+      then [ (concatStringsSep " " (
+          ["Match"]
+          ++ optional (cf.host != null)          "host \"${cf.host}\""
+          ++ optional (cf.exec != null)          "exec \"${cf.exec}\""
```

Isn't `cf.exec` guaranteed to be not null here?

```suggestion
          ["Match"]
          ++ optional (cf.host != null)          "host \"${cf.host}\""
          ++ ["exec \"${cf.exec}\""]
```

---

There are other `Match` conditions as well.
This is a state, in which adding support for them is easier, because one would only need to add it to the condition and to the list.
However, I haven't committed that yet, since I haven't implemented the checks needed for the other conditions yet.